src/js/components/table.js

```diff
@@ -0,0 +1,230 @@
+const select = require("../utils/select");
+const behavior = require("../utils/behavior");
+const { CLICK } = require("../events");
+const { prefix: PREFIX } = require("../config");
+
+const TABLE = `.${PREFIX}-table`;
+const SORTED = "aria-sort";
+const ASCENDING = "ascending";
+const DESCENDING = "descending";
+const SORT_OVERRIDE = "data-sort-value";
+
+const ICON_SOURCE = `
+  <svg class="${PREFIX}-icon" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
+    <g id="descending" fill="transparent">
+      <path d="M19,15l-1.41-1.41L13,18.17V2H11v16.17l-4.59-4.59L5,15l7,7L19,15z"/>
+    </g>
+    <g id="ascending" fill="transparent">
+      <path transform="rotate(180, 12, 12)" d="M19,15l-1.41-1.41L13,18.17V2H11v16.17l-4.59-4.59L5,15l7,7L19,15z"/>
+    </g>
+    <g id="unsorted" fill="transparent">
+      <polygon points="15.17 15 13 17.17 13 6.83 15.17 9 16.58 7.59 12 3 7.41 7.59 8.83 9 11 6.83 11 17.17 8.83 15 7.42 16.41 12 21 16.59 16.41 15.17 15"/>
+    </g>
+  </svg>
+`;
+
+const SORT_BUTTON_CLASS = `${PREFIX}-table__header__button`;
+const SORT_BUTTON = `.${SORT_BUTTON_CLASS}`;
+const SORTABLE_HEADER = `th[data-sortable]`;
+const ANNOUNCEMENT_REGION = `.${PREFIX}-table__announcement-region[aria-live="polite"]`;
+
+
+/** Gets the data-sort-value attribute value, if provided — otherwise, gets
+ * the innerText or textContent — of the child element (HTMLTableCellElement)
+ * at the specified index of the given table row
+ *
+ * @param {number} index
+ * @param {array<HTMLTableRowElement>} tr
+ * @return {boolean}
+ */
+const getCellValue = (tr, index) => tr.children[index].getAttribute(SORT_OVERRIDE)
+  || tr.children[index].innerText
+  || tr.children[index].textContent;
+
+/**
+ * Compares the values of two row array items at the given index, then sorts by the given direction
+ * @param {number} index
+ * @param {string} direction
+ * @return {boolean}
+ */
+const compareFunction = (index, direction) => (a, b) => ((v1, v2) =>
+    // if neither value is empty, and if both values are already numbers, compare numerically. Otherwise, compare alphabetically based on current user locale
+    v1 !== '' && v2 !== '' && !Number.isNaN(Number(v1)) && !Number.isNaN(Number(v2)) ? v1 - v2 : v1.toString().localeCompare(v2, navigator.language, {numeric: true, ignorePunctuation: true})
```

Comment:
Can we clean this up or format it in a way that's more readable?

Reply:
Made some clarifying changes -- what do you think? https://github.com/uswds/uswds/pull/3950/commits/540972fdc9e9abe550bbf25181a1bd77971a1991

Reply:
Yeah that and running prettier via `npx prettier --write ./src/js/**/*.js` might help.